ontology/modbus.ttl

```diff
@@ -0,0 +1,358 @@
+@prefix : <https://www.w3.org/2019/wot/modbus#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xml: <http://www.w3.org/XML/1998/namespace> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix schema: <http://schema.org/> .
+@prefix vann: <http://purl.org/vocab/vann/> .
+@prefix dct: <http://purl.org/dc/terms/> .
+@prefix binding: <https://www.w3.org/2019/wot/binding#> .
+@base <https://www.w3.org/2019/wot/modbus> .
+
+@prefix om: <http://www.ontology-of-units-of-measure.org/resource/om-2/>.
+	
+
+
+<> 	rdf:type owl:Ontology ;					
+	rdfs:label "Modbus in RDF"@en;
+	rdfs:comment "Vocabulary to represent Modbus messages in RDF."@en ;
+	vann:preferredNamespacePrefix "modv" ;
+   	vann:preferredNamespaceUri : ;
+	dct:author _:jakob;
+	dct:contributor _:jakob .
+
+_:jakob a schema:Person ;
+	schema:name "Jakob Müller" .
+
+	
+	
+	
+	 
+	### Classes
+	
+	###  http://www.example.com/ns/modbus#Message
+	
+	:Message rdf:type owl:Class;
+	    	rdfs:isDefinedBy <> ;
+        	rdfs:comment "A Modbus Message"@en ;
+			rdfs:label "A Modbus Message"@en.
+	
+	
+	###  http://www.example.com/ns/modbus#Request
+	
+	:Request rdf:type owl:Class ;
+			 rdfs:comment "Request"@en ;
+ 			 rdfs:label "Request"@en ;
+
+			 rdfs:subClassOf :Message .
+	
+	###  http://www.example.com/ns/modbus#Response
+	
+	:Response rdf:type owl:Class ;
+				rdfs:comment "Response"@en ;
+				rdfs:label "Response"@en ;
+
+			  rdfs:subClassOf :Message .
+	
+	
+	###  http://www.example.com/ns/modbus#Code
+	
+	:Code rdf:type owl:Class;
+					rdfs:comment "Code"@en .
+	
+	:FunctionCode rdf:type owl:Class ;
+				rdfs:comment "Function Code"@en ;
+				rdfs:label "Function Code"@en ;
+
+				rdfs:subClassOf :Code, schema:Enumeration .
+
+	:FunctionName rdf:type owl:Class ;
+				rdfs:comment "A Modbus human readable function name"@en ;
+				rdfs:label "Function Name"@en ;
+				rdfs:subClassOf schema:Enumeration .
+
+	:Entity rdf:type owl:Class ;
+				rdfs:comment "A Modbus registry type"@en ;
+				rdfs:label "Entity"@en ;
+				rdfs:subClassOf schema:Enumeration .
+	
+	:ResponseCode rdf:type owl:Class ;
+				  rdfs:comment "Response Code"@en ;
+				  rdfs:label "Response Code"@en ;
+
+				  rdfs:subClassOf :Code, schema:Enumeration  .
+	
+	
+	### Properties
+	
+	# ###  http://www.example.com/ns/modbus#options
+	
+	:function   rdf:type owl:ObjectProperty ;

```

Comment:
I would recommend to introduce functionCode and functionName. One of them can be used. Alternatively, both can be used at the same time, but should not contradict each other.